src/helpers.js

```diff
@@ -14,11 +14,30 @@ function hashFromString(path, separator, hash) {
     var obj     = tmp_obj;
```

Comment:
Actually I'm fairly certain we do, without it, tmp_obj gets overwritten with nested values so we wind up returning only part of the intended obj. 40 tests fail by removing this and returning tmp_obj instead.